### PR TITLE
Expose tail in experiment configuration

### LINKED ISSUE #1851: Expose tail in configuration
**Context**
The OAT implementation for sensitivity analysis has a variable `tail` that dictates the magnitude of the perturbation. It would be nice to make this an argument to the algorithm and expose it as a configurable option when running sensitivity analysis.

**Solution**
I would like to be able to write something like

```
type: sensitivity
algorithm: one-at-a-time
tail: 0.99
```

### PR DESCRIPTION
**Pull request by marenwb**
**Issue**
Resolves #1851

**Approach**
- Added `tail: Optional[float]` to `Experiment Config`
- Added `tail`as parameter to `one-at-the-time()`
- Made 0.99 the default tail

### CODE REVIEW

**Comment by marenwb** [comment]
Is it correct that the tail can be in the interval (0,1)?

File: ert3/algorithms/_sensitivity.py [hunk 1]
```diff
@@ -11,19 +11,22 @@ def _build_base_records(
 
 
 def one_at_the_time(
-    parameters: MutableMapping[str, Distribution]
+    parameters: MutableMapping[str, Distribution], tail: Optional[float] = None
```
Comment (markusdregi) [inline ert3/algorithms/_sensitivity.py:14]: Why not do the following instead?

```suggestion
    parameters: MutableMapping[str, Distribution], tail: float = 0.99
```

File: ert3/config/_experiment_config.py [hunk 2]
```diff
@@ -22,15 +23,19 @@ class Config:
 class ExperimentConfig(_ExperimentConfig):
     type: Literal["evaluation", "sensitivity"]
     algorithm: Optional[Literal["one-at-a-time"]]
+    tail: Optional[float] = None
```
Comment (markusdregi) [inline ert3/config/_experiment_config.py:26]: Same here. It is better to require it to be a float and default it to `0.99`

File: ert3/engine/_run.py [hunk 3]
```diff
@@ -150,14 +150,17 @@ def _load_ensemble_parameters(
...
     parameters_config: ert3.config.ParametersConfig,
     workspace_root: pathlib.Path,
     experiment_name: str,
 ) -> None:
     parameter_distributions = _load_ensemble_parameters(
         ensemble, parameters_config, workspace_root
     )
-    input_records = ert3.algorithms.one_at_the_time(parameter_distributions)
+    input_records = ert3.algorithms.one_at_the_time(
+        parameter_distributions, experiment_config.tail
```
Comment (markusdregi) [inline ert3/engine/_run.py:162]: It is good practise to pass keyword arguments with name, like: `tail=experiment_config.tail`

File: tests/ert_tests/ert3/config/test_experiment_config.py [hunk 4]
```diff
@@ -10,11 +10,16 @@ def test_valid_evaluation():
     assert experiment_config.algorithm == None
 
 
-def test_valid_sensitivity():
-    raw_config = {"type": "sensitivity", "algorithm": "one-at-a-time"}
+@pytest.mark.parametrize(
```
Comment (markusdregi) [inline tests/ert_tests/ert3/config/test_experiment_config.py:13]: We should also have a test ensuring that passing no tail also is a valid configuration.

**Review by markusdregi — COMMENTED** [review]
The implementation looks good 👍🏻 I suggest we default the tail to a number instead of `None` both in the algorithm and in the configuration.

I think this work would split nicely over two commits:
- Expose tail in one-at-the-time algorithm
- Make tail configurable in one-at-the-time experiments

The first commit should then get dedicated tests verifying the impact of changing the tail.

File: ert3/config/_experiment_config.py (hunk 2 above)
Comment (marenwb) [reply ert3/config/_experiment_config.py:26]: I think there should be no default value. With a default value the evaluation experiment will also have a tail that is `0.99`, which does not make sense. So just `tail: Optional[float]`.

Comment (markusdregi) [reply ert3/config/_experiment_config.py:26]: You are right :)

**Review by markusdregi — APPROVED** [review]
Thanks for addressing the comments! This PR looks good 👏🏻

**Pull request merged**